drivers/dac/dac_gd32.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2021 BrainCo Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT gd_gd32_dac
+
+#include <errno.h>
+#include <drivers/pinctrl.h>
+#include <drivers/dac.h>
+#include <soc.h>
+
+#define LOG_LEVEL CONFIG_DAC_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(dac_gd32);
+
+/**
+ * Most of GD32 series SOCs' DAC have 2 output, DAC0 and DAC1.
+ * But some SOC, like GD32F350 only have one output, DAC0.
+ */
+#if !defined(CONFIG_SOC_GD32F350)
+#define DAC_GD32_OUT_NUM  2
+#define DAC_GD32_OUT_DUAL 1
+#else
+#define DAC_GD32_OUT_NUM  1
+#endif /* !CONFIG_SOC_GD32F350 */
+
+#define DAC_GD32_DEFAULT_VALUE 0
+
+struct dac_gd32_config {
+	uint32_t reg;
+	uint32_t rcu_periph_clock;
+	const struct pinctrl_dev_config *pcfg;
+};
+
+struct dac_gd32_data {
+	uint8_t resolutions[DAC_GD32_OUT_NUM];
+};
+
+static void dac_gd32_write(struct dac_gd32_data *data,
+		    uint8_t dacx, uint32_t value)
```

Comment:
```c
static void dac_gd32_write(struct dac_gd32_data *data,
			   uint8_t dacx, uint32_t value)
```
make sure your editor uses 8sp width for tabs

Reply:
done.